gadget/src/lib.rs

```diff
@@ -34,12 +38,55 @@ pub async fn ignite_lc_relayer(ctx: LightClientRelayerContext) -> anyhow::Result
 			},
 		};
 		let api_client = Arc::new(api_client);
-		let pair = std::fs::read_to_string(&ctx.lc_init_config.path_to_signer_secret_key)
-			.expect("failed to read secret key");
-		let pair = subxt::ext::sp_core::sr25519::Pair::from_string(&pair, None);
+		// Read path first will change to string later for parsing
+		let path = Path::new(&ctx.lc_init_config.path_to_signer_secret_key);
+		// Check if the file exists, its permissions, and if it's empty
+        let pair = if path.exists() {
```

Comment:
to be honest, I would just parsed it like we did but just trimmed string before subxt::ext::sp_core::sr25519::Pair::from_string(&pair, None);

Reply:
Trimming can be a quick solution but this does handle crazy newlines etc, I had tested with a few different chopped up keys from the documentation in the repo. I agree I can make most of this another function vs cluttering the init logic. 
I do agree its not as simplistic as I would like but were tackling quite a few scenarios and providing the user better feedback